oc-organisation-picker.html

```diff
@@ -2,200 +2,215 @@
+                    permission: String,
+                    /**
+                     * The selected organisation
+                     */
+                    selectedOrganisation: {
+                        type: Number,
+                        statePath: 'organisation.id'
+                    },
+                    /**
+                     * The default organisation ID
+                     */
+                    __defaultOrganisationId: Number,
+                    /**
+                     * flag if the user has been signed in
+                     */
+                    _signedIn : {
+                        type: Boolean,
+                        statePath: 'identity.loggedIn',
+                        observer: '_loadOrganisations'
+                    },
+                    /**
+                     * The list of organisations the user can choose from
+                     */
+                    organisations: {
+                        type: Array,
+                        notify: true
+                    },
+                };
             }
 
-            this.$.authApi.getOrganisations(permission)
-                .then(function(request) {
-                    this.organisations = request.response.results
-                        .sort(function(a, b) {
-                            var nameA = a.name.toUpperCase(); // ignore upper and lowercase
-                            var nameB = b.name.toUpperCase();
-                            if (nameA < nameB) {
-                                return -1;
-                            }
-                            if (nameA > nameB) {
-                                return 1;
-                            }
-                            return 0;
-                        });
-                }.bind(this))
-                .catch(function() {
-                    this.errorMessage = 'Failed to load organisations';
-                }.bind(this));
-        }
+            static get observers() {
+                return [
+                    '_initSelectedOrganisation(organisations.*)',
+                ]
+            }
 
-        _loadDefaultOrganisation() {
-            this.defaultOrganisationId = this.defaultOrganisation ? Number(this.defaultOrganisation) : null;
-        }
+            ready() {
+                super.ready();
+            }
 
-        _initSelectedOrganisation(organisations) {
-            if (!organisations.base || organisations.base.length === 0) {
-                this.errorMessage = 'No organisations';
-                return;
+            /**
+             * Load the organisations where the current user has the given permission.
+             * @param {String} permission
+             */
+            _loadOrganisations(signedIn) {
+
+                if (!signedIn) {
+                    this.organisations = null;
+                    return;
+                }
+
+                let permission = 'bWFuYWdlbWVudF9jb25zb2xlOnZpZXc6b3JnYW5pc2F0aW9u';
```

Comment:
We should start think of removing the permission hashes out of te front-end code.